File: FinanceGrid.WebClient/src/models/stocks/user-stock-watch.ts
import { Share } from "../shares/share";

export class UserStockWatch {
    id!: string;
    userEmail!: string;
    listName!: string;
    stockSymbol!: string;
    note?: string;
    purchaseGuidToShares!: { [purchaseGuid: string]: Share };
}

File: StocksApi/StocksApi.Abstractions/Models/Shares/UserStockWatch.cs
namespace StocksApi.Abstractions.Models.Shares
{
    public class UserStockWatch
    {
        public string? Id { get; set; }
        public required string UserEmail { get; set; }
        public required string ListName { get; set; }
        public required string StockSymbol { get; set; }
        public string? Note { get; set; }
        public required Dictionary<string, Share> PurchaseGuidToShares { get; set; }
    }
}
